### PR TITLE
add torch.no_grad to kfserver

### LINKED ISSUE #87: Models with calibration don't work in Kubeflow
From the logs:
```
[E 210204 01:39:56 web:1789] Uncaught exception POST /v1/models/x40org-thingpedia-models-contex78tualx2fen:predict (127.0.0.1)
    HTTPServerRequest(protocol='http', host='x40org-thingpedia-models-contex78tualx2fen-predictor-default.almond-staging.svc.cluster.local', method='POST', uri='/v1/models/x40org-thingpedia-models-contex78tualx2fen:predict', version='HTTP/1.1', remote_ip='127.0.0.1')
    Traceback (most recent call last):
      File "/usr/local/lib64/python3.8/site-packages/tornado/web.py", line 1704, in _execute
        result = await result
      File "/usr/local/lib/python3.8/site-packages/kfserving/handlers/http.py", line 59, in post
        response = (await model.predict(request)) if inspect.iscoroutinefunction(model.predict) else model.predict(request)
      File "/opt/genienlp/genienlp/kfserver.py", line 53, in predict
        results = self.server.handle_request(request)
      File "/opt/genienlp/genienlp/server.py", line 96, in handle_request
        output = generate_with_model(self.model, [batch], self.numericalizer, task, self.args,
      File "/opt/genienlp/genienlp/validate.py", line 117, in generate_with_model
        confidence_scores = confidence_estimator.estimate(confidence_features)
      File "/opt/genienlp/genienlp/calibrate.py", line 428, in estimate
        features, labels = self.convert_to_dataset(confidences, train=False)
      File "/opt/genienlp/genienlp/calibrate.py", line 423, in convert_to_dataset
        features = self.convert_to_features(confidences, train)
      File "/opt/genienlp/genienlp/calibrate.py", line 359, in convert_to_features
        features = TreeConfidenceEstimator._concatenate(features)
      File "/opt/genienlp/genienlp/calibrate.py", line 345, in _concatenate
        concat = np.concatenate([features_list[j][i] for j in range(len(features_list))])
      File "<__array_function__ internals>", line 5, in concatenate
      File "/usr/local/lib64/python3.8/site-packages/torch/tensor.py", line 630, in __array__
        return self.numpy()
    RuntimeError: Can't call numpy() on Tensor that requires grad. Use tensor.detach().numpy() instead.
[E 210204 01:39:56 web:2239] 500 POST /v1/models/x40org-thingpedia-models-contex78tualx2fen:predict (127.0.0.1) 26735.81ms
```

This prevents deploying recently trained models because they are all calibrated.

### PR DESCRIPTION
**Pull request by jgd5**
Fixes #87

### CODE REVIEW

**Review by gcampax — APPROVED** [review]
Yeah. I checked why difference with genienlp server and it makes sense we need this.

**Review by s-jse — COMMENTED** [review]
This fixes the problem. A cleaner one would be moving `with torch.no_grad():` from `server.run()` to inside `server.handle_request()`.

**Comment by jgd5** [comment]
> This fixes the problem. A cleaner one would be moving `with torch.no_grad():` from `server.run()` to inside `server.handle_request()`.

Good idea.

File: genienlp/kfserver.py [hunk 1]
```diff
@@ -31,6 +31,7 @@
 import logging
 
 import kfserving
+import torch
```
Comment (s-jse) [inline genienlp/kfserver.py]: Leftover import!

**Review by s-jse — APPROVED** [review]

**Pull request merged**